fennel/datasets/test_invalid_dataset.py

```diff
@@ -220,6 +221,77 @@ class RatingActivity:
     t: datetime
 
 
+def test_incorrect_assign_expr_type():
```

Comment:
Could you plz add one test case for filter as well where the expression doesn't evaluate to bool?